### PR TITLE
add cart items to cart page, add full functionality to cart

### PR DESCRIPTION
**Pull request by olena-ole**

### CODE REVIEW

File: src/components/Card.tsx [hunk 1]
```diff
@@ -44,7 +44,17 @@ export const Card = ({ product }: Props) => {
...
-    dispatch(addItemToCart({ id: product.id, count: 1 }));
+    dispatch(
+      addItemToCart({
+        id: product._id,
+        name: product.name,
+        price: product.priceDiscount
+          ? product.priceDiscount
+          : product.priceRegular,
+        image: product.images[0],
+        count: 1,
```
Comment (MaxSchmide) [inline src/components/Card.tsx:55]: move this data to variable and pass this variable as prop

File: src/components/CartItem.tsx [hunk 2]
```diff
@@ -1,41 +1,49 @@
...
-  const [itemCount, setItemCount] = useState(1);
+import { removeFromCart, increaseItemCount, decreaseItemCount } from '../redux';
+import { ICartItem } from '../types/CartItem';
+
+type Props = {
+  item: ICartItem;
+};
+
+const CartItem: React.FC<Props> = ({ item }) => {
+  const { name: itemName, price, image, count, id } = item;
```
Comment (MaxSchmide) [inline src/components/CartItem.tsx:14]: rename it ti product (optional)

File: src/components/CartItem.tsx [hunk 3]
```diff
@@ -1,41 +1,49 @@
...
+
   const maxItems = 5;
   const minItems = 1;
```
Comment (MaxSchmide) [inline src/components/CartItem.tsx:18]: why do we need max and min for number of products in cart?

File: src/pages/CartPage.tsx [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
-
-const totalSum = '2657';
-const itemsAmount = '3';
+import CartItem from '../components/CartItem';
+import { getTotalProductsCost } from '../utils/getTotalCost';
+import { getTotalItemsCount } from '../utils/getTotalItemsCount';
 
 export const CartPage: React.FC = () => {
+  const cartItems = useSelector((state: RootState) => state.cart.items);
+
```
Comment (MaxSchmide) [inline src/pages/CartPage.tsx:12]: 1. Don't use default useSelector, we have our own useAppSelector which already typified
2. (optional) destructure it like { items } from state.cart

**Review by MaxSchmide — CHANGES_REQUESTED** [review]

File: src/components/CartItem.tsx (hunk 2 above)
Comment (olena-ole) [reply src/components/CartItem.tsx:14]: we already have interface IProduct, so I'll leave it as it is

Comment (MaxSchmide) [reply src/components/CartItem.tsx:14]: it's a prop) u can name it as u wish

**Review by MaxSchmide — APPROVED** [review]
looks good

File: src/components/CartItem.tsx (hunk 2 above)
Comment (olena-ole) [reply src/components/CartItem.tsx:14]: ok, I thought about ICartItem))

**Review by ihorhladkov — APPROVED** [review]

**Pull request merged**